server/account/src/operations.ts

```diff
@@ -58,26 +59,68 @@ import { accountPlugin } from './plugin'
 const WORKSPACE_COLLECTION = 'workspace'
 const ACCOUNT_COLLECTION = 'account'
 const INVITE_COLLECTION = 'invite'
-
 /**
  * @public
  */
 export const ACCOUNT_DB = 'account'
 
-const getEndpoint = (): string => {
-  const endpoint = getMetadata(toolPlugin.metadata.Endpoint)
-  if (endpoint === undefined) {
+/**
+ * Returns a hash code for a string.
+ * (Compatible to Java's String.hashCode())
+ *
+ * The hash code for a string object is computed as
+ *     s[0]*31^(n-1) + s[1]*31^(n-2) + ... + s[n-1]
+ * using number arithmetic, where s[i] is the i th character
+ * of the given string, n is the length of the string,
+ * and ^ indicates exponentiation.
+ * (The hash value of the empty string is zero.)
+ *
+ */
+function hashWorkspace (workspace: Workspace): number {
+  return [...workspace.workspace].reduce((hash, c) => (Math.imul(31, hash) + c.charCodeAt(0)) | 0, 0)
+}
+
+export enum EndpointKind {
+  Internal,
+  External
+}
+
+const getEndpoint = (ctx: MeasureContext, workspaceInfo: Workspace, kind: EndpointKind): string => {
+  const transactorsUrl = getMetadata(accountPlugin.metadata.Transactors)
+  if (transactorsUrl === undefined) {
     throw new Error('Please provide transactor endpoint url')
   }
-  return endpoint
-}
+  const endpoints = transactorsUrl.split(',')
 
-const getTransactor = (): string => {
-  const transactor = getMetadata(toolPlugin.metadata.Transactor)
-  if (transactor === undefined) {
-    throw new Error('Please provide transactor url')
+  if (endpoints.length === 0) {
```

Comment:
this condition is inaccurate
when the separator is not an empty string, then `split` never returns  an empty array. There will be at least one element, but the element might be an empty string. Probably need something like this:
```
const endpoints = transactorsUrl.split(',').filter((p) => p.length > 0)
```